src/main/java/net/minecraftforge/fluids/TileFluidHandler.java

```diff
@@ -31,15 +33,19 @@ public void writeToNBT(NBTTagCompound tag)
 
     /* IFluidHandler */
     @Override
-    public int fill(EnumFacing from, FluidStack resource, boolean doFill)
+    public int fill(@Nullable EnumFacing from, FluidStack resource, boolean doFill)
     {
-        return tank.fill(resource, doFill);
+        if(from==null)
+        	from=EnumFacing.NORTH;
+    	return tank.fill(resource, doFill);
```

Comment:
Mixed spaces/tabs, you have them all over the files. With proper whitespace lines like these wouldn't even appear in the diff.